gateway/src/apicast/policy/websocket/websocket.lua

```diff
@@ -0,0 +1,23 @@
+-- This policy enables Web socket pass through on APIcast
+
+local _M = require('apicast.policy').new('Websocket', 'builtin')
+
+local new = _M.new
+
+function _M.new(configuration)
+  local policy = new(configuration)
+  return policy
+end
+
+local function is_websocket_connection()
+  local headers = ngx.req.get_headers()
+  return headers["Upgrade"] ~= nil and headers["Sec-WebSocket-Key"] ~= nil
+end
+
+function _M:rewrite()
+  if is_websocket_connection() then
+    ngx.var.upstream_connection_header = "Upgrade"
```

Comment:
Isn't the caller responsible for setting this header?
https://developer.mozilla.org/en-US/docs/Web/HTTP/Protocol_upgrade_mechanism#Upgrading_to_a_WebSocket_connection

Reply:
It's reset on apicast.conf and this is to make sure that we set to Upgrade.

Reply:
but why is this header not set by the application that calls apicast?

Reply:
Is set, but is overwritten by default here: 

https://github.com/3scale/APIcast/blob/master/gateway/conf.d/apicast.conf#L88

So, the changes allow only to change that, just because is a websocket server.